src/bin/tikv-ctl.rs

```diff
@@ -21,25 +21,415 @@ extern crate clap;
+                    if start_ts.map_or(true, |ts| lock_info.get_lock_version() == ts) {
+                        // FIXME: "lock type" is lost in kvproto.
+                        let pk = escape(lock_info.get_primary_lock()).into_bytes();
+                        let k = escape(lock_info.get_key()).into_bytes();
+                        lock_info.set_primary_lock(pk);
+                        lock_info.set_key(k);
+                        println!("\tlock cf value: {}", print_to_string(&lock_info));
+                    }
+                }
+                if cfs.contains(&CF_DEFAULT) {
+                    for mut value_info in mvcc.take_values().into_iter() {
+                        if commit_ts.map_or(true, |ts| value_info.get_ts() == ts) {
+                            let value = escape(value_info.get_value()).into_bytes();
+                            value_info.set_value(value);
+                            println!("\tdefault cf value: {}", print_to_string(&value_info));
+                        }
+                    }
+                }
+                if cfs.contains(&CF_WRITE) {
+                    for write_info in mvcc.take_writes().into_iter() {
+                        if start_ts.map_or(true, |ts| write_info.get_start_ts() == ts) &&
+                            commit_ts.map_or(true, |ts| write_info.get_commit_ts() == ts)
+                        {
+                            // FIXME: short_value is lost in kvproto.
+                            println!("\t write cf value: {}", print_to_string(&write_info));
+                        }
+                    }
+                }
+                println!("");
+                future::ok::<(), String>(())
+            },
+        );
+        if let Err(e) = scan_future.wait() {
+            eprintln!("{}", e);
+            process::exit(-1);
+        }
+    }
+
+    fn diff_region(
+        &self,
+        region: u64,
+        db: Option<&str>,
+        raft_db: Option<&str>,
+        host: Option<&str>,
+    ) {
+        let rhs_debug_executor = new_debug_executor(db, raft_db, host);
+
+        let r1 = self.get_region_info(region);
+        let r2 = rhs_debug_executor.get_region_info(region);
+        println!("region id: {}", region);
+        println!("db1 region state: {:?}", r1.region_local_state);
+        println!("db2 region state: {:?}", r2.region_local_state);
+        println!("db1 apply state: {:?}", r1.raft_apply_state);
+        println!("db2 apply state: {:?}", r2.raft_apply_state);
+
+        match (r1.region_local_state, r2.region_local_state) {
+            (None, None) => return,
+            (Some(_), None) | (None, Some(_)) => {
+                println!("db1 and db2 don't have same region local_state");
+                return;
+            }
+            (Some(region_local_1), Some(region_local_2)) => {
+                let start_key = keys::data_key(region_local_1.get_region().get_start_key());
```

Comment:
Can it be checked by `==`?

Reply:
Compare two `metapb.Region` will also compare them `peers`, but here we only need to compare data in them.

Reply:
I think they should be same if two regions is going to be the same.